compound.hh

```diff
@@ -174,29 +184,40 @@ public:
         }
         const value_type& operator*() const { return _current; }
         const value_type* operator->() const { return &_current; }
-        bool operator!=(const iterator& i) const { return _v.begin() != i._v.begin(); }
-        bool operator==(const iterator& i) const { return _v.begin() == i._v.begin(); }
+        bool operator==(const iterator& i) const { return _remaining == i._remaining; }
     };
-    static iterator begin(const bytes_view& v) {
+    static iterator begin(managed_bytes_view v) {
         return iterator(v);
     }
-    static iterator end(const bytes_view& v) {
+    static iterator end(managed_bytes_view v) {
         return iterator(typename iterator::end_iterator_tag(), v);
     }
-    static boost::iterator_range<iterator> components(const bytes_view& v) {
+    static boost::iterator_range<iterator> components(managed_bytes_view v) {
         return { begin(v), end(v) };
     }
-    value_type deserialize_value(bytes_view v) const {
+    value_type deserialize_value(managed_bytes_view v) const {
         std::vector<bytes> result;
         result.reserve(_types.size());
         std::transform(begin(v), end(v), std::back_inserter(result), [] (auto&& v) {
-            return bytes(v.begin(), v.end());
+            return to_bytes(v);
         });
         return result;
     }
+    bool less(managed_bytes_view b1, managed_bytes_view b2) const {
```

Comment:
@michoecho ping, I skimmed over the new version and didn't see a change w.r.t. this.

Reply:
Sorry, I didn't answer because I thought that this was a thought-provoking comment for the future, rather than a change request :P
The `less()`, `equal()`, etc. in `types.hh` work on `abstract_type`, and the `less()`, `equal()`, etc. in `compound.hh` work on `compound_type`. They do different things, and they aren't duplicates, if that's what you were concerned about. In fact, the `compound_type` versions call the `abstract_type` versions sometimes (if the compound is not trivially comparable), but sometimes not (if the compound be compared with a simple `memcmp()`).
To move them to a common place, we would have to unify `compound_type` and `abstract_type`, but we don't want to, I think?

Reply:
Ah, I thought they are doing the same thing, the `managed_bytes_view` arguments (that all versions have) mislead me.